dist/pythonlibs/testrunner/spawn.py

```diff
@@ -0,0 +1,62 @@
+# Copyright (C) 2018-19 Freie Universität Berlin
+#               2017 Cenk Gündoğan <cenk.guendogan@haw-hamburg.de>
+#               2016 Kaspar Schleiser <kaspar@schleiser.de>
+#               2014 Martine Lenders <mlenders@inf.fu-berlin.de>
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+import os
+import pexpect
+import signal
+import subprocess
+import time
+from traceback import extract_tb
+
+PEXPECT_PATH = os.path.dirname(pexpect.__file__)
+RIOTBASE = (os.environ.get('RIOTBASE') or
+            os.path.abspath(os.path.join(os.path.dirname(__file__),
+                                         "..", "..", "..")))
+
+# Setting an empty 'TESTRUNNER_START_DELAY' environment variable use the
+# default value (3)
+MAKE_TERM_STARTED_DELAY = int(os.environ.get('TESTRUNNER_START_DELAY') or 3)
+
+
+def list_until(l, cond):
+    return l[:([i for i, e in enumerate(l) if cond(e)][0])]
+
+
+def find_exc_origin(exc_info):
+    pos = list_until(extract_tb(exc_info),
+                     lambda frame: frame[0].startswith(PEXPECT_PATH)
+                     )[-1]
+    return (pos[3], os.path.relpath(os.path.abspath(pos[0]), RIOTBASE), pos[1])
+
+
+def setup_child(timeout=10, spawnclass=pexpect.spawnu, env={}, logfile=None):
```

Comment:
This one is a dangerous pattern. Because if anything modifies `env` in the function, it updates the default argument for all the other calls.
Also giving `{}` means give no environment which is different from giving the default `None`.
The documentation does not explicit it but it is the case.

If I remove the `os.environ` in `run` it fails with as `test` is using `native`.

```
BOARD=iotlab-m3 make -C tests/bloom_bytes/ flash test
...
Terminal program /home/harter/work/git/RIOT/tests/bloom_bytes/bin/native/tests_bloom_bytes.elf is required but not found in PATH.  Aborting.
/home/harter/work/git/RIOT/tests/bloom_bytes/../../Makefile.include:540: recipe for target 'term' failed
make: *** [term] Error 1
Unexpected end of file in expect script at "child.expect_exact("Testing Bloom filter.")" (tests/bloom_bytes/tests/01-run.py:17)

Process already stopped
/home/harter/work/git/RIOT/tests/bloom_bytes/../../Makefile.include:568: recipe for target 'test' failed
make: *** [test] Error 1
make: Leaving directory '/home/harter/work/git/RIOT/tests/bloom_bytes'
```

```suggestion
def setup_child(timeout=10, spawnclass=pexpect.spawnu, env=None, logfile=None):
```